kyuubi-server/web-ui/src/views/editor/components/Editor.vue

```diff
@@ -0,0 +1,273 @@
+        name="result">
+        <Result :data="sqlResult" />
+      </el-tab-pane>
+    </el-tabs>
+  </div>
+</template>
+
+<script lang="ts" setup>
+  import MonacoEditor from '@/components/monaco-editor/index.vue'
+  import Result from './Result.vue'
+  import { ref, reactive, onUnmounted, toRaw } from 'vue'
+  import type { Ref } from 'vue'
+  import * as monaco from 'monaco-editor'
+  import { format } from 'sql-formatter'
+  import { ElMessage } from 'element-plus'
+  import { useI18n } from 'vue-i18n'
+  import { getEngineType } from '@/utils/engine'
+  import {
+    openSession,
+    closeSession,
+    runSql,
+    getSqlRowset,
+    getSqlMetadata,
+    getLog
+  } from '@/api/editor'
+  import type { IResponse, ISqlResult, IFields, ILog } from './types'
+
+  const { t } = useI18n()
+  const param = reactive({
+    engineType: 'SPARK_SQL'
+  })
+  const limit = ref(10)
+  const sqlResult: Ref<any[] | null> = ref(null)
+  const sqlLog = ref('')
+  const activeTab = ref('result')
+  const resultLoading = ref(false)
+  const logLoading = ref(false)
+  const sessionIdentifier = ref('')
+  const theme = ref('customTheme')
+  const editorVariables = reactive({
+    editor: {} as any,
+    language: 'sql',
+    content: '',
+    options: {}
+  })
+
+  const editorMounted = (editor: monaco.editor.IStandaloneCodeEditor) => {
+    editorVariables.editor = editor
+  }
+  const handleFormat = () => {
+    toRaw(editorVariables.editor).setValue(
+      format(toRaw(editorVariables.editor).getValue())
+    )
+  }
+
+  const editorSave = () => {
+    handleFormat()
+  }
+
+  const handleContentChange = (value: string) => {
+    editorVariables.content = value
+  }
+
+  const handleQuerySql = async () => {
+    resultLoading.value = true
+    logLoading.value = true
+    const openSessionResponse: IResponse = await openSession({
+      'kyuubi.engine.type': param.engineType
+    }).catch(errorCatch)
+    if (!openSessionResponse) return
+    sessionIdentifier.value = openSessionResponse.identifier
+
+    const runSqlResponse: IResponse = await runSql(
+      {
+        statement: editorVariables.content,
+        runAsync: false
+      },
+      sessionIdentifier.value
+    ).catch(errorCatch)
+    if (!runSqlResponse) return
+
+    Promise.all([
+      getSqlRowset({
+        operationHandleStr: runSqlResponse.identifier,
+        fetchorientation: 'FETCH_NEXT',
+        maxrows: limit.value
+      }),
+      getSqlMetadata({
+        operationHandleStr: runSqlResponse.identifier
+      })
+    ])
+      .then((result: any[]) => {
+        sqlResult.value = result[0]?.rows?.map((row: IFields) => {
+          const map: { [key: string]: any } = {}
+          row.fields?.forEach(({ value }: ISqlResult, index: number) => {
+            map[result[1].columns[index]?.columnName] = value
+          })
+          return map
+        })
+      })
+      .catch(() => {
+        ElMessage({
```

Comment:
Use `errorCatch` instead

Reply:
Hi, thanks for review. errorCatch has its own logic like set sqlResult and sqlLog to empty, not very suitable to use it here.

Reply:
Thanks for your comment. In my opinion, when we run this SQL and encounter an error, we already have both set the result to an empty array and used the el-message component to notify the user. However, should we also try to obtain the SQL log from the error message if it is available?

Reply:
@zwangsheng 
Certainly, got your point, what you think is the api error message might be helpful to users, we also need it!

If we want to display all the error info for the APIs through the log, there might be some issues here:
The current design for executing SQL is as follows:
* First, we call the openSession API, which returns the sessionIdentifier.
* Then insert the returned sessionIdentifier into the second API, runSql, which returns the runSqlIdentifier.
These two steps must be executed synchronously because they are dependent on each other.

After the above steps, there are two more steps remaining:
* Retrieving the SQL's return results (here, there are two APIs, one for getting the table's field names, and one for getting the table's field values).
* Getting the SQL log.
Here, considering the user experience, we don't need to make the APIs execute synchronously. That is, getting the SQL results and getting the SQL log will be sent asynchronously to the service, as the SQL log API doesn't need to wait for the SQL result, and the SQL result API doesn't need to wait for the SQL log.

Now, here's the issue:
Because the last three APIs are asynchronous, if we only overwrite their error information into the log, it might provide a poor user experience. For example, suppose the getSqlResult API returns an error in 30 seconds (it could be two errors), and it logs the error, then user is watching the log, but suddenly the log gets overwritten because at 45 seconds, the getSqlLog also sends logs. Appending the log to the API isn't good enough either because the log display might not be very clear.

**If this error log is useful for user, can we consider this design below?**
* The log section will only display query log information. 
* Error messages will be placed in the error box so that users can clearly see the error content and list error messages from multiple APIs as a reference.  (see screenshot)

Thanks.

Reply:
<img width="1094" alt="image" src="https://github.com/apache/kyuubi/assets/13530192/14f67c22-2f23-422e-bad2-971a3db493f8">

Reply:
Thanks for your long answer, SGTM, it's fine for me to see server query log in log item and other api error log in el-message.

Reply:
<img width="1646" alt="image" src="https://github.com/apache/kyuubi/assets/13530192/f3044728-d881-4162-9cd3-da7874c61c73">

Reply:
<img width="1646" alt="image" src="https://github.com/apache/kyuubi/assets/13530192/5c24f327-48d1-48b2-a46b-1a718e9536ac">

Reply:
Hi @zwangsheng 
Has added new error messages for single or multiple errors, please help review, thanks!

Reply:
@tongwl could you put the Run button at the head of the line? and use a light color for it?

Reply:
Also, please use `SQL` instead of `Sql` in the message.

Reply:
Hi @pan3793 , sure, got!

Reply:
<img width="1682" alt="image" src="https://github.com/apache/kyuubi/assets/13530192/5ecc70a2-dbc7-47e8-91f4-b81938371a43">
<img width="1397" alt="image" src="https://github.com/apache/kyuubi/assets/13530192/6fe8adc8-02c2-4c64-b0ea-d24b997fa821">
Hi @pan3793 , please help review again!

Reply:
<img width="1686" alt="image" src="https://github.com/apache/kyuubi/assets/13530192/23eeb2ed-e7fd-4952-bb37-7a3c84ae9ac6">